File: sdk/python/src/rocksky/resources/like.py
"""``app.rocksky.like.*`` — like / dislike songs and shouts."""

from __future__ import annotations

from typing import Any

from ._base import Resource


class LikeResource(Resource):
    async def like_song(self, uri: str) -> dict[str, Any]:
        return await self._transport.procedure(
            "app.rocksky.like.likeSong", body={"uri": uri}, auth=True
        ) or {}

    async def dislike_song(self, uri: str) -> dict[str, Any]:
        return await self._transport.procedure(
            "app.rocksky.like.dislikeSong", body={"uri": uri}, auth=True
        ) or {}

    async def like_shout(self, uri: str) -> dict[str, Any]:
        return await self._transport.procedure(
            "app.rocksky.like.likeShout", body={"uri": uri}, auth=True
        ) or {}

    async def dislike_shout(self, uri: str) -> dict[str, Any]:
        return await self._transport.procedure(
            "app.rocksky.like.dislikeShout", body={"uri": uri}, auth=True
        ) or {}

File: sdk/ruby/lib/rocksky/resources/like.rb
module Rocksky
  module Resources
    # `app.rocksky.like.*` endpoints. All require an authenticated client.
    class Like < Base
      # Like a song.
      def like_song(uri:)
        procedure("app.rocksky.like.likeSong", body: { uri: uri })
      end

      # Remove a like on a song.
      def dislike_song(uri:)
        procedure("app.rocksky.like.dislikeSong", body: { uri: uri })
      end

      # Like a shout.
      def like_shout(uri:)
        procedure("app.rocksky.like.likeShout", body: { uri: uri })
      end

      # Remove a like on a shout.
      def dislike_shout(uri:)
        procedure("app.rocksky.like.dislikeShout", body: { uri: uri })
      end
    end
  end
end
